### ISSUE TITLE
Issue with loading the products fixture data on other than localhost

### ISSUE THREAD

**Issue by boboci9** [reactioncommerce/reaction#457]
I found a new issue with the loading of the fixture data since the merge where you took out the domains from the private/data/Shops.json
In the loading of the fixtures you have
```
Fixtures.loadData ReactionCore.Collections.Shops
  Fixtures.loadData ReactionCore.Collections.Products
  Fixtures.loadData ReactionCore.Collections.Tags
  Fixtures.loadI18n ReactionCore.Collections.Translations

  # if ROOT_URL update shop domain
  # for now, we're assuming the first domain is the primary
  currentDomain = Shops.findOne().domains[0]
  if currentDomain isnt getDomain()
    ReactionCore.Events.info "Updating domain to " + getDomain()
    Shops.update({domains:currentDomain},{$set:{"domains.$":getDomain()}})
```
But the products can not be loaded without the shopId and the `@shopId` won't be found only after the domains are set for the shop, therefore this version returns an error when trying to load save the fixture product with Shop is missing
as a quick fix I tried adding the domain setup after the shop fixture load and before the products and the error was gone, product was inserted
```
  Fixtures.loadData ReactionCore.Collections.Shops

  # if ROOT_URL update shop domain
  # for now, we're assuming the first domain is the primary
  currentDomain = Shops.findOne().domains[0]
  if currentDomain isnt getDomain()
    ReactionCore.Events.info "Updating domain to " + getDomain()
    Shops.update({domains:currentDomain},{$set:{"domains.$":getDomain()}})

  Fixtures.loadData ReactionCore.Collections.Products
  Fixtures.loadData ReactionCore.Collections.Tags
  Fixtures.loadI18n ReactionCore.Collections.Translations
```

**Comment by aaronjudd** [comment]
@boboci9 @jshimko can you test for this in the next couple builds?  I'm pretty sure what @boboci9 is reporting is the same problem we were seeing -> products, tags not loading, or loading prior to shop being created. I worked on this issue in the beta refactor (see changes to fixtures.js), and was able to replicate, and then remove the problem - but this has the feel of a ghost in the machine as well - so I'd live to get some verification that it's resolved.

**Comment by jshimko** [comment]
@boboci9 This should be all set now (in the development branch of reaction-core).  I just added a check for the Shops fixtures to be ready.

```js
  Fixtures.loadData(ReactionCore.Collections.Shops);

  // start checking once per second if Shops collection is ready,
  // then load the rest of the fixtures when it is
  let wait = Meteor.setInterval(function () {
    if (!!ReactionCore.Collections.Shops.find().count()) {
      Meteor.clearInterval(wait);
      Fixtures.loadData(ReactionCore.Collections.Products);
      Fixtures.loadData(ReactionCore.Collections.Tags);
      Fixtures.loadI18n(ReactionCore.Collections.Translations);
    }
  }, 1000);
```
[Commit](https://github.com/reactioncommerce/reaction-core/commit/a5cffcd6699f77769dd60b0512431d1a197c6877)

Let me know if this solves it on your end too.  If you use Docker, you can also try the latest development build here [`reactioncommerce/prequel:latest`](https://hub.docker.com/r/reactioncommerce/prequel/)

**Comment by kefirchik** [comment]
I have the same problems yesterday and now it is solved. :smile:

**Comment by boboci9** [comment]
Sorry I couldn't check the new version yet, I have some important deadlines, but I will start digging deep as soon as possible.
